tests/lib/rules/no-var.js

```diff
@@ -36,6 +36,12 @@ ruleTester.run("no-var", rule, {
 				parserOptions: { ecmaFeatures: { globalReturn: true } },
 			},
 		},
+		{
+			code: "declare global { var bar: 'car' }",
+			languageOptions: {
+				parser: require("@typescript-eslint/parser"),
+			},
+		},
```

Comment:
What we have been following up untill now for TS cases is creating a separate `ruleTesterTypescript` instance. You can refer this PR https://github.com/eslint/eslint/pull/19560/

https://github.com/eslint/eslint/blob/58a171e8f0dcc1e599ac22bf8c386abacdbee424/tests/lib/rules/no-loss-of-precision.js#L322-L327